### PR TITLE
feat(filmstrip-pagination): mobile support.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Comment by saghul** [comment]
> @saghul I've rebased the PR and fixed the bounce issue and the issue with the 2 columns on 5th participant. I think we should be in pretty good shape for merging but let me know if you want me to do more changes!

Great to hear! I'll take a look on Monday and merge it if all checks out!

File: react/features/filmstrip/components/native/Thumbnail.js [hunk 1]
```diff
@@ -255,21 +330,28 @@ function _mapStateToProps(state, ownProps) {
...
     const renderModeratorIndicator = !_isEveryoneModerator
-        && participant && participant.role === PARTICIPANT_ROLE.MODERATOR;
+        && participant?.role === PARTICIPANT_ROLE.MODERATOR;
+    const participantInLargeVideo = id === largeVideo.participantId;
 
     return {
         _audioMuted: audioTrack?.muted ?? true,
-        _largeVideo: largeVideo,
+        _isScreenShare: isScreenShare,
+        _isFakeParticipant: participant?.isFakeParticipant,
```
Comment (saghul) [inline react/features/filmstrip/components/native/Thumbnail.js:345]: nit: alpha sort

File: react/features/filmstrip/components/native/TileView.js [hunk 2]
```diff
@@ -64,100 +72,138 @@ type Props = {
...
     /**
-     * Implements React's {@link Component#componentDidMount}.
+     * The styles for the FlatList.
+     */
+    _flatListStyles: Object;
+
+    /**
+     * The styles for the content container of the FlatList.
+     */
+    _contentContainerStyles: Object;
```
Comment (saghul) [inline react/features/filmstrip/components/native/TileView.js:101]: nit: alpha-sort

File: react/features/filmstrip/middleware.native.js [hunk 3]
```diff
@@ -2,27 +2,32 @@
...
+import { setTileViewDimensions } from './actions';
 import { updateRemoteParticipants, updateRemoteParticipantsOnLeave } from './functions';
 import './subscriber';
 
 /**
  * The middleware of the feature Filmstrip.
  */
 MiddlewareRegistry.register(store => next => action => {
+    if (action.type === PARTICIPANT_LEFT) {
+        updateRemoteParticipantsOnLeave(store, action.participant?.id);
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/filmstrip/middleware.native.js:21]: 👍

**Review by saghul — COMMENTED** [review]
Left a couple of style nits. I'm going to test this now.

File: react/features/filmstrip/components/native/TileView.js (hunk 2 above)
Comment (hristoterezov) [reply react/features/filmstrip/components/native/TileView.js]: Fixed

File: react/features/filmstrip/components/native/Thumbnail.js (hunk 1 above)
Comment (hristoterezov) [reply react/features/filmstrip/components/native/Thumbnail.js:344]: Fixed

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Fantastic work @hristoterezov ! 🚀 🚀 🚀

**Comment by saghul** [comment]
I realize I missed your long comment 🤦‍♀️ I'll keep an eye on the debounce thing, but if you tested it already hopefully we're good there!

**Pull request merged**